stats/stats/src/data_source/kinds/local_db/parameters/update/batching/parameters/mod.rs

```diff
@@ -60,3 +60,40 @@ where
         Ok(found)
     }
 }
+
+/// Batch update "step" that clears all data for this chart
+/// and inserts the newly queried data.
+///
+/// Since it removes all data each time, it makes sense to
+/// use it only with max batch size.
+pub struct ClearAllAndPassStep;
+
+impl<Resolution> BatchStepBehaviour<Resolution, Vec<TimespanValue<Resolution, String>>, ()>
+    for ClearAllAndPassStep
+where
+    Resolution: Timespan + Clone + Send + Sync,
+{
+    async fn batch_update_values_step_with(
+        db: &DatabaseConnection,
+        chart_id: i32,
+        _update_time: DateTime<Utc>,
+        min_blockscout_block: i64,
+        _last_accurate_point: TimespanValue<Resolution, String>,
+        main_data: Vec<TimespanValue<Resolution, String>>,
+        _resolution_data: (),
+    ) -> Result<usize, ChartError> {
+        clear_all_chart_data(db, chart_id)
+            .await
+            .map_err(ChartError::StatsDB)?;
+        PassVecStep::batch_update_values_step_with(
+            db,
+            chart_id,
+            _update_time,
+            min_blockscout_block,
+            _last_accurate_point,
+            main_data,
+            _resolution_data,
+        )
+        .await
```

Comment:
so, if `batch_update_values_step_with` fail, we just cleared all data? maybe.. db.begin()?

Reply:
I agree